### ISSUE TITLE
DeepLIFT with rules

### ISSUE THREAD

**Issue by annasli378** [hans66hsu/nn_interpretability#6]
Hi,

I would like to use DeepLIFT for my finetuned models (Resnet, Densenet etc.), however it was only working with DeepLIFTRules.NoRule.
My dataset consist of RGB images (254, 254, 3). In example provided everything was working smoothly, but for my model, when I try using any Rules it give me errors.

For my densenet201 model with 37 classes:
```
interpreter = DeepLIFT(model, [str(i) for i in range(37)], None, DeepLIFTRules.RevealCancel)
for data in dataloaders['test']:
            image, label = data
            out = interpreter.interpret(image)
```

I got error:
```
  File "...\nn_interpretability_main\nn_interpretability\interpretation\deeplift\deeplift.py", line 252, in forward_hook
    self.inputs.append(input[0].data.clone().to(device))

AttributeError: 'list' object has no attribute 'data'
```

It is possible that the problem is with my models? Can I even run DeepLIFT with them for RGB images?

**Comment by hans66hsu** [comment]
RGB images shouldn't be a problem. One possible issue I speculate from your error is that the image you provided is not a Tensor. [14.Demo.ipynb](https://github.com/hans66hsu/nn_interpretability/blob/main/14.Demo.ipynb) contains an example of using DeepLIFT to interpret a prediction of an RGB image.

**Comment by annasli378** [comment]
Thank you for your answer,

I was using this example and it works perfectly on vgg, squeezenet and alexnet. However, I tried also googlenet, resnet101, densenet201 and inceptionV3 and for all of them its giving different errors so maybe there is a problem with those architectures? I tried both pretreined imported models and my models (finetuned) as well.

I know there is problem with resnet and DeepLIFT method implemented in the Captum repository, so maybe those errors are due to the architecture?
